### ISSUE TITLE
Quota enforcements in runtime and admin as per plan

### ISSUE THREAD

**Issue by pjain1** [rilldata/rill#5220]
- Data size quota enforcements
- Trail period end enforcements on API call. UI can block the page by checking subscription details but API level enforcement need to be added ?
- Plan metadata may change, so periodically sync changes or we may want to just provide more quotas for a POC (both can be achieved by creating one-off plans ?). We may need to enable some features for poc for customers (not sure about how to achieve this feature based enablement)

**Comment by pjain1** [comment]
At risk if https://github.com/rilldata/rill-private-issues/issues/493 takes time

**Comment by pjain1** [comment]
> Data size quota enforcements

We have decided to use rill metrics for this ? We will improve heartbeat emission to ask for current data size directly from the instance olap engine. For duckdb it can be just size project dir, for clickhouse and druid, we will query relevant systems table to get this information and emit to rill-metrics. rill metrics project already ingests org table from postgres which has the quotas information. Now we can create an alert that will compare the actual usage values with the quota and send alerts to us internally and then we can take action.

> Plan metadata may change, so periodically sync changes or we may want to just provide more quotas for a POC (both can be achieved by creating one-off plans ?). We may need to enable some features for poc for customers (not sure about how to achieve this feature based enablement)

For this we will go with creating one-off plans, details in this notion [doc]( https://www.notion.so/rilldata/Orb-Plans-Runbook-c57133e70d874ec991bc97c665c399f4?pvs=4#8439cec76c0c4b89b1b08fca356456f1)

> Trail period end enforcements on API call. UI can block the page by checking subscription details but API level enforcement need to be added ?

For this we will create a background job that runs everyday or so to check orgs whose trial period is going to end soon log an error or warning that datadog can alert on internally, we can extend this to send email notifications further.

**Comment by pjain1** [comment]
For this ticket I think there may be only two immediate items -
1. If we change quota for an existing plan then implement an API that fetches the new quotas and updates corresponding orgs. For one-off overrides for POC, we will just create a new without disturbing existing plans.
2. Background jobs that runs everyday or so to check orgs whose trial period is going to end soon and logs an error or warning that datadog can alert on internally.

**Comment by pjain1** [comment]
For both we need an API to directly fetch customer subscribed to particular a plan or one has to list all customers and check subscription on each customer. Don't see any such API, have asked in Orb channel.

**Comment by pjain1** [comment]
Trial end check worker - https://github.com/rilldata/rill/pull/5316

**Comment by pjain1** [comment]
dashboard for alerting - https://github.com/rilldata/rill-cloud-metrics/pull/20

**Comment by pjain1** [comment]
We will use this dashboard for alerting internally, once this PR is merged, we can create alerts. Only thing is Issues with creating alert with 1 hour interval through UI, discussion here - https://rilldata.slack.com/archives/C02T907FEUB/p1722501690356229

**Comment by pjain1** [comment]
Here's the updated dashboard PR that uses the no spiky metric - https://github.com/rilldata/rill-cloud-metrics/pull/21. Once this is merged and alert is setup through UI then we can close this issue. If we want hard fail then that can be done in separate issue.
